dev/tests/integration/testsuite/Magento/AdvancedPricingImportExport/_files/create_products.php

```diff
@@ -4,26 +4,45 @@
  * See COPYING.txt for license details.
  */
 
-$productModel = \Magento\TestFramework\Helper\Bootstrap::getObjectManager()
-    ->create(\Magento\Catalog\Model\Product::class);
+declare(strict_types=1);
 
-$productModel->setTypeId(\Magento\Catalog\Model\Product\Type::TYPE_SIMPLE)
+use Magento\Catalog\Api\ProductRepositoryInterface;
+use Magento\Catalog\Model\Product;
+use Magento\Catalog\Model\Product\Attribute\Source\Status;
+use Magento\Catalog\Model\Product\Type;
+use Magento\Catalog\Model\Product\Visibility;
+use Magento\Framework\ObjectManagerInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+
+/** @var ObjectManagerInterface $objectManager */
+$objectManager = Bootstrap::getObjectManager();
+
+/**
+ * @var Product $productModel
+ * @var ProductRepositoryInterface $productRepository
+ */
+$productModel = $objectManager->create(Product::class);
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
+
+$productModel->setTypeId(Type::TYPE_SIMPLE)
     ->setAttributeSetId(4)
     ->setName('AdvancedPricingSimple 1')
     ->setSku('AdvancedPricingSimple 1')
     ->setPrice(321)
-    ->setVisibility(\Magento\Catalog\Model\Product\Visibility::VISIBILITY_BOTH)
-    ->setStatus(\Magento\Catalog\Model\Product\Attribute\Source\Status::STATUS_ENABLED)
+    ->setVisibility(Visibility::VISIBILITY_BOTH)
+    ->setStatus(Status::STATUS_ENABLED)
     ->setWebsiteIds([1])
     ->setCategoryIds([])
     ->setStockData(['qty' => 100, 'is_in_stock' => 1, 'manage_stock' => 1])
-    ->setIsObjectNew(true)
-    ->save();
+    ->setIsObjectNew(true);
+
+$productRepository->save($productModel);
 
 $productModel->setName('AdvancedPricingSimple 2')
     ->setId(null)
-    ->setUrlKey(null)
+    ->setUrlKey('')
```

Comment:
Why you changed type null to empty string here?

Reply:
Hi @ihor-sviziev  
I see that we shoud set `string` for `setUrlKey`
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Catalog/Model/Product.php#L32
Thanks

Reply:
@Usik2203 theoretically it's true, but practically you've changed type of argument, that might lead to generating incorrect url key or not generating url key at all.
I think it should be better to create separate object and not reuse $productModel there, so you'll not need to run setUrlKey at all there.

What do you thing?